src/main/java/org/springframework/data/elasticsearch/core/routing/DefaultRoutingResolver.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.routing;
+
+import org.springframework.data.elasticsearch.core.mapping.SimpleElasticsearchMappingContext;
+import org.springframework.data.elasticsearch.core.mapping.SimpleElasticsearchPersistentEntity;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * Default implementation of the {@link RoutingResolver} interface. Retuns {@literal null} for the non-bean method and
+ * delegates to the corresponding persistent entity for the bean-method.
+ * 
+ * @author Peter-Josef Meisch
+ * @since 4.2
+ */
+public class DefaultRoutingResolver implements RoutingResolver {
+
+	private final SimpleElasticsearchMappingContext mappingContext;
+
+	public DefaultRoutingResolver(MappingContext mappingContext) {
```

Comment:
Parameter should make use of generics.

Reply:
changing this to `MappingContext<? extends ElasticsearchPersistentEntity<?>, ElasticsearchPersistentProperty>` is no different than using `SimpleElasticsearchMappingContext`. So I'd rather change the constructor parameter to `SimpleElasticsearchMappingContext` and remove the assert.

If we use `MappingContext<?,?>` we need a cast/check in `getRouting` for the class of the persistent entity. As `resolveRouting(bean)` is defined on `ElasticsearchPersistentEntity`. Using a `SimpleElasticsearchMappingContext` the return type of `getPersistentEntity()` is `SimpleElasticsearchPersistentEntity` and that's ok.

Reply:
done